### ISSUE TITLE
Put reference in its own element (bibliography)

### ISSUE THREAD

**Issue by nvitucci** [asciidoctor/asciidoctor#4548]
I think this is an Asciidoctor request rather than a language one, but I'm happy to ask this in the language repository instead.

At the moment, when converted to HTML, the reference to an entry is rendered as part of the `<p>` element. From the examples in the [docs](https://docs.asciidoctor.org/asciidoc/latest/sections/bibliography/), the References section is converted as follows:

```
<div class="ulist bibliography">
  <ul class="bibliography">
    <li>
      <p><a id="pp"></a>[pp] Andy Hunt &amp; Dave Thomas. The Pragmatic Programmer: From Journeyman to Master. Addison-Wesley. 1999.</p>
    </li>
  </ul>
</div>
```

Can we have the reference in its own element? For example, if `[pp]` were converted to `<span>[pp]</span>` or something similar, it could easily be styled with CSS. I believe it could even just be text for the currently empty `<a>` element, unless you have excluded this for another reason.

**Comment by mojavelinux** [comment]
Actually, this would be a change to consider for the next gen HTML converter. We aren't making any more substantial structural changes to the existing HTML converter as that would severely impact backwards compatibility. But that's why we are rethinking the whole thing. So I'm going to classify this issue as part of the next gen effort.

**Comment by mojavelinux** [comment]
I don't think this would be a language issue (so this is the right place). This is really about how the information from the semantic element is organized in the converted output. At the moment, that is up to the AsciiDoc processor. In the future, there may be an HTML converter specification, but we think that will come out of the next gen converter, not the current one (which is intentionally backwards compatible with the HTML that has been produced from AsciiDoc for almost two decades).

**Comment by nvitucci** [comment]
Thank you for the clarification. Feel free to close it then.

**Comment by mojavelinux** [comment]
I didn't say anything about closing it. Rather, I added it for
consideration to the next gen converter.

Keep in mind you can extend the converter if this is a customization you
need urgently.
